### ISSUE TITLE
Support Connect and Connect Amp

### ISSUE THREAD

**Issue by 889laurent** [duncan3dc/sonos#7]
i tried the las version end with this code
```php
require __DIR__ . "/vendor/autoload.php";
use duncan3dc\Sonos\Network;
$sonos = new Network (new \Doctrine\Common\Cache\ArrayCache);

$tests = $sonos->getSpeakers();
foreach ($tests as $test) {
    echo $test->name . " " .  $test->room . " " . $test->getVolume() . "% "  .$$
}
```
there are not find the sonos connect and sonos connect amp in my network.

**Comment by duncan3dc** [comment]
Unfortunately those devices aren't supported at the moment (as I don't have them to test with).

If you have their ip addresses then you could manually create a `Device` instance and pass soap calls to it:
```php
$amp = new Device("192.168.0.66");
$volume = (int) $amp->soap("RenderingControl", "GetVolume", [
    "Channel"   =>  "Master",
]);
echo "The volume is currently: {$volume}\n";
```

**Comment by 889laurent** [comment]
Thanks for your answer. In your précédent version (patch in my rpi) is ok ?
If you want I try in my network or better tried with ssh on my network


Laurent MARIO Directeur Technique

  [cid:logo2015989526]
________________________________

tel fixe: +33 4 93 23 27 27 télécopie: +33 4 93 23 27 20  tel direct: +33 4 93 23 27 21 mobile: +33 6 83 81 26 35 mail: laurent.MARIO@valtinee.fr

Route Départementale 2205 Lieu dit la Sorbière 06420 SAINT SAUVEUR SUR TINEE

________________________________

Ce mail a été envoyé depuis la societé VALTINEE. Si ce mail ne vous concerne pas, merci de le renvoyer a son expéditeur et de le détruire. L'ensemble de ce mail et de ses pièces jointes sont la propriété exclusive de l'entreprise VALTINEE. Tout usage non autorisé est strictement interdit.

Notre Entreprise, en complément de sa certification ISO 9001 (qualité) et ISO 14001 (environement), a été certifiée pour sa démarche sécurité OHSAS 18001  par CERTIBAT<http://www.certibat.fr>

Nous adhérons aux fédérations du TP<http://www.fntp.fr/> ,du batiment<http://www.ffbatiment.fr/> et du syndicat du  Béton Prêt a l'emploi<http://www.snbpe.org/>


[cid:picture198d9c3]                                                                                                      [cid:devdurable12e3f0]

 Issa Nissa<http://www.ogcnice.fr/> (*Allez NICE)                                  Pensez à l'environnement et n'imprimez ce mail que si nécessaire.



De : Craig Duncan [mailto:notifications@github.com]
Envoyé : lundi 2 mars 2015 10:06
À : duncan3dc/sonos
Cc : laurent MARIO
Objet : Re: [sonos] not find sonos connect AMP et sonos connect (#7)


Unfortunately those devices aren't supported at the moment (as I don't have them to test with).

If you have their ip addresses then you could manually create a Device instance and pass soap calls to it:

$amp = new Device("192.168.0.66");

$volume = (int) $amp->soap("RenderingControl", "GetVolume", [

    "Channel"   =>  "Master",

]);

echo "The volume is currently: {$volume}\n";

**Comment by duncan3dc** [comment]
The previous version considered all devices as Speakers. But the Connect Amp doesn't support all the same features as speakers, which caused the error you reported in #5.

I will try and find some time to log on to your pi and do some testing, thanks

**Comment by 889laurent** [comment]
the précédent version as you modified on my rpi via tts works with sonos connect amp ans sonos connect. (but not présent after update).The ip of connect amp ans connect is 192.168.0.60 ("Piscine") and 192.168.0.38 ("Cuisine ete")

**Comment by 889laurent** [comment]
in my case I added the ZP90 and ZP120 in device.php and it works
line 119 device.php  return in_array($model, ["S1", "S3", "S5", "ZP90", "ZP120"], true);

**Comment by 889laurent** [comment]
i have buy a player and i want to pilot with pour api. Please give me the reference of model (ex "S1,...
thanks

**Comment by 889laurent** [comment]
its a playbar , sorry

**Comment by 889laurent** [comment]
I searched and I found! S9 is

**Comment by duncan3dc** [comment]
Great, thanks @889laurent I will add support for the playbar shortly

**Comment by duncan3dc** [comment]
Playbar support is available in 1.1.3

**Comment by 889laurent** [comment]
the playbar reads the digital input when switching on the TV. The playback status is "playing"! How do I know if it is playing music from the queue or from digital input? It'll be interesting to add a read status Audio Input or radio

**Comment by duncan3dc** [comment]
Unfortunately I don't have a Playbar or Play:5 to test digital input at the moment. Once I do I will add support for it

**Comment by 889laurent** [comment]
thanks for your help. If you want i tried your code on my playbar for test .

**Comment by zoic21** [comment]
Is there any new of suppoer Connect and Connect AMP ? How can I help you to support those ?

**Comment by duncan3dc** [comment]
Unfortunately I have no plans to purchase a Connect or Connect AMP, so support for these is unlikely.

I will be working on support for other inputs (TV, stereo, etc) from Play:5 PlayBar. After I purchase a Play:5.

What other features to Connect or Connect:AMP offer?

**Comment by zoic21** [comment]
For now nothings work on connect, when I tried play fonction I can an error tht say it's not a speaker. I used 1.1.3 version, I tried with 1.2.X dev version but same error : No speaker found for the IP address

**Comment by zoic21** [comment]
Finally a find how, I just add "ZP90" => "Connect" to Device.php : 
----
	public function isSpeaker() {
		$model = $this->getModel();

		$models = [
			"S1" => "PLAY:1",
			"S3" => "PLAY:3",
			"S5" => "PLAY:5",
			"S9" => "PLAYBAR",
			"ZP90" => "Connect",
		];

		return in_array($model, array_keys($models), true);
	}
----

And my connect work perfectly

**Comment by duncan3dc** [comment]
The connect is not added there as @889laurent reported in issue #5 that it gave him an error.

As I don't have the devices I'm not really bothered either way. If you 2 would like ZP90 and ZP120 to be listed as valid speakers then I'll do that, can you both confirm you want this? Thanks

**Comment by duncan3dc** [comment]
This is available in 1.1.4

**Comment by zoic21** [comment]
Thank it's work
